config/src/providers/remappings.rs

```diff
@@ -12,6 +12,35 @@ use std::{
 };
 use tracing::trace;
 
+/// Helper type for extendable remappings
+/// that can only be extended with remappings that don't overlap with existing ones.
+type Remappings = Vec<Remapping>;
+
+/// Helper trait for extendable remappings where the remappings can only be extended with
+/// remappings that don't overlap with existing ones.
+trait ExtendableRemappings {
+    fn extend_remappings<T>(&mut self, remappings: T)
+    where
+        T: IntoIterator<Item = Remapping>;
```

Comment:
the new type should just have a push function which only pushes if not contains, 

and then extend which calls push for each entry in the iter